homeassistant/components/unifi/sensor.py

```diff
@@ -185,3 +236,172 @@ async def options_updated(self) -> None:
         if not self.controller.option_allow_uptime_sensors:
             await self.remove_item({self.client.mac})
+
+
+@dataclass
+class UniFiDeviceSensorEntityDescription(SensorEntityDescription):
+    """Describes UniFi device sensor entities."""
+
+    enabled: str | None = None
+    value_fn: Callable[[UniFiDevice], Any] | None = None
+
+    def get_value(self, device: UniFiDevice) -> Any:
+        """Return value from UniFi device."""
+        if self.value_fn is not None:
+            return self.value_fn(device)
+
+        return None
+
+    def is_enabled(self, device: UniFiDevice) -> bool:
+        """Return whether the entity should be enabled."""
+        if self.enabled is None:
+            return True
+
+        return device.raw.get(self.enabled, False)
+
+
+DEVICE_SENSORS: tuple[UniFiDeviceSensorEntityDescription, ...] = (
+    UniFiDeviceSensorEntityDescription(
+        key=CPU_UTILIZATION_SENSOR,
+        name="CPU Utilization",
+        native_unit_of_measurement=PERCENTAGE,
+        icon="mdi:speedometer",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        enabled="system-stats",
+        value_fn=lambda device: float(device.raw.get("system-stats", {}).get("cpu")),
+    ),
+    UniFiDeviceSensorEntityDescription(
+        key=MEMORY_UTILIZATION_SENSOR,
+        name="Memory Utilization",
+        native_unit_of_measurement=PERCENTAGE,
+        icon="mdi:memory",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        enabled="system-stats",
+        value_fn=lambda device: float(device.raw.get("system-stats", {}).get("mem")),
+    ),
+    UniFiDeviceSensorEntityDescription(
+        key=UPTIME_SENSOR,
+        name="Uptime",
+        icon="mdi:clock",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda device: dt_util.now() - timedelta(seconds=device.raw["uptime"]),
+    ),
+    UniFiDeviceSensorEntityDescription(
+        key=TEMPERATURE_SENSOR,
+        name="Temperature",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        enabled="has_temperature",
+        value_fn=lambda device: device.raw["general_temperature"],
+    ),
+    UniFiDeviceSensorEntityDescription(
+        key=CPU_TEMPERATURE_SENSOR,
+        name="CPU Temperature",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        enabled="temperatures",
+        value_fn=lambda device: next(
+            (
+                temperature["value"]
+                for temperature in device.raw.get("temperatures", [])
```

Comment:
Can we expand with more temperature attributes in aiounifi?

Reply:
We can. I wanted to avoid having dependent PRs as I wasn't sure if you would want these changes or not. I have some other changes to make it `aiounifi` and was going to do this then. Or if you prefer I can do the `aiounifi` changes first and make this PR depend on those.

Reply:
I'm starting my vacation time now so I think I will have some time to continue my refactoring of aiounifi as well, but lets make small PRs to aiounifi that we can release with only changes relevant to this PR.